### PR TITLE
Apply Promise to methods in LedgerStorage and test code

### LINKED ISSUE #69: Improve by using Promise for methods of LedgerStorage
Using async and await, we can easily write repeating sentences without using the nested callback.
But to take advantage of this, we must define the `Promise<return type>` as a return function.
But the methods of LedgerStorage are not. 
These are modified as functions that return `Promise<return type>`.
This will provide various choices in the use of these methods.

Example of use of `async / await / Promis`

``` TypeScript
async function putAllBlockData (ledger_storage: LedgerStorage,
    callback: (err: Error | null) => void)
{
    // Use Promise to record block data synchronously.
    let save = (storage: LedgerStorage, block: any) =>
    {
        return new Promise<boolean>((resolve, reject) => {
            storage.putBlocks(block, ()=>{resolve(true);}, (error)=>{reject(error);});
        });
    };

    for (let elem of sample_data)
    {
        try
        {
            await save(ledger_storage, elem);
        }
        catch (error)
        {
            callback(error);
            return;
        }
    }
    callback(null);
}
```

### PR DESCRIPTION
**Pull request by MichaelKim20**
I applied Promise not to use the nested callback.
I applied it to the method of LedgerStorage first.
And I applied this to the test code.

Linked #69

### CODE REVIEW

**Comment by TrustHenry** [comment]
I think we should add a function description of the Promise return value.
To help users understand the get functions or functions,

File: src/Stoa.ts [hunk 1]
```diff
@@ -22,7 +22,6 @@ class Stoa {
             if (err != null)
             {
                 console.error(err);
-                throw new Error(err.message);
```
Comment (Geod24) [inline src/Stoa.ts]: Why would the ctor not throw if there's an error ?

**Comment by Geod24** [comment]
Love it !

**Comment by TrustHenry** [comment]
There are test codes that have not been applied due to the new code merge.
Could you Apply `Promise` it together?
https://github.com/bpfkorea/stoa/blob/42970f554626279046b0bfc84e904d3c47fa8c7c/tests/Storage.test.ts#L359

File: src/modules/storage/LedgerStorage.ts [hunk 2]
```diff
@@ -33,11 +33,8 @@ export class LedgerStorage extends Storages
 
     /**
      * Creates tables related to the ledger.
-     * @param callback If provided, this function will be called when
-     * the database was finished successfully or when an error occurred.
-     * The first argument is an error object.
```
Comment (Geod24) [inline src/modules/storage/LedgerStorage.ts:34]: Need to document the return value now

File: src/modules/storage/LedgerStorage.ts [hunk 3]
```diff
@@ -183,21 +179,16 @@ export class LedgerStorage extends Storages
     /**
      * Gets a block data
      * @param height the height of the block to get
-     * @param onSuccess This function will be called when
-     * the database was finished successfully
-     * @param onError This function will be called when
-     * an error occurred.
```
Comment (Geod24) [inline src/modules/storage/LedgerStorage.ts:186]: Ditto documentation on return value

**Comment by Geod24** [comment]
Missing a lot of doc upgrades. I  commented on 2 occurrences, but there are much more that need fixing.

File: src/Stoa.ts (hunk 1 above)
Comment (MichaelKim20) [reply src/Stoa.ts]: It's my mistake. I added it again.

**Comment by MichaelKim20** [comment]
@Geod24 I rebased it, and I added a document.

**Review by TrustHenry — APPROVED** [review]
LGTM

**Comment by MichaelKim20** [comment]
> LGTM

Thank you for review

**Comment by MichaelKim20** [comment]
I rebased it

**Pull request merged**